model/src/main/scala/hmda/model/institution/Institution.scala

```diff
@@ -0,0 +1,54 @@
+package hmda.model.institution
+
+/**
+ * A financial institution, geared towards requirements for filing HMDA data.
+ */
+case class Institution(
+    id: Int,
```

Comment:
So this seems to be very similar to the previous implementation [here](https://github.com/cfpb/hmda-platform/blob/master/model/src/main/scala/hmda/model/fi/Institution.scala). Of course, this one is more complete and I favor it. But the other one needs to be removed. 
Also, the previous version is used in the HTTP API, i.e. [here](https://github.com/cfpb/hmda-platform/blob/master/api/src/main/scala/hmda/api/http/InstitutionsHttpApi.scala#L44). 

This needs to be consolidated. I'm ok with creating new issues that address this separately from this PR